tests/test_base.py

```diff
@@ -1,23 +1,74 @@
 import pytest
 from speckle.objects.base import Base
+from speckle.api import operations
+from contextlib import ExitStack as does_not_raise
 
 
-class TestBase:
-    def test_empty_prop_names(self):
-        base = Base()
-        with pytest.raises(ValueError):
-            base[""] = "empty"
-        with pytest.raises(ValueError):
-            base["@"] = "empty"
+@pytest.mark.parametrize(
+    "invalid_prop_name",
+    [
+        (""),
+        ("@"),
+        ("@@wow"),
+        ("this.is.bad"),
+        ("super/bad"),
+    ],
+)
+def test_empty_prop_names(invalid_prop_name: str) -> None:
+    base = Base()
+    with pytest.raises(ValueError):
+        base[invalid_prop_name] = "🐛️"
 
-    def test_invalid_prop_names(self):
-        base = Base()
 
-        with pytest.raises(ValueError):
-            base["@@wow"] = "bad"
+class FakeModel(Base):
+    """Just a test class type."""
 
-        with pytest.raises(ValueError):
-            base["this.is.bad"] = "also bad"
+    foo: str = ""
 
-        with pytest.raises(ValueError):
-            base["super/bad"] = "no no no"
+
+def test_new_type_registration() -> None:
+    """Test if a new subclass is registered into the type register."""
+    assert Base.get_registered_type("FakeModel") == FakeModel
+    assert Base.get_registered_type("🐺️") is None
```

Comment:
as per the other comment, this should return `Base` rather than `None`

Reply:
Im gonna add a test for deserialization to check the fallback.